File: onion-architecture/java/src/main/java/com/example/api/EnrolmentController.java
package com.example.api;

import com.example.application.EnrolmentApplicationService;
import org.springframework.web.bind.annotation.*;

@RestController
@RequestMapping("/enrolments")
public class EnrolmentController {

    private final EnrolmentApplicationService service;

    public EnrolmentController(EnrolmentApplicationService service) {
        this.service = service;
    }

    @PostMapping
    public void enrol(@RequestParam Long studentId, @RequestParam Long courseId) {
        service.enrol(studentId, courseId);
    }
}

File: onion-architecture/kotlin/src/main/kotlin/com/example/api/EnrolmentController.kt
package com.example.api

import com.example.application.EnrolmentApplicationService
import org.springframework.web.bind.annotation.*

@RestController
@RequestMapping("/enrolments")
class EnrolmentController(
    private val service: EnrolmentApplicationService
) {

    @PostMapping
    fun enrol(@RequestParam studentId: Long, @RequestParam courseId: Long) {
        service.enrol(studentId, courseId)
    }
}